Various fixes and improvements in chat domain

djing-chan opened the pull request.

axpoems reviewed (changes requested):
NACK

I made a quick review, here are some comments:

* I am getting NPE the first time I open a chat domain. See stack trace below.

```
Dec-18 14:55:12.377 [JavaFX Application Thread] ERROR b.d.DesktopController: Uncaught exception from thread Thread[JavaFX Application Thread,5,main] 
Dec-18 14:55:12.377 [JavaFX Application Thread] ERROR b.d.DesktopController: Uncaught exception: java.lang.NullPointerException: Cannot invoke "javafx.scene.Scene.getWindow()" because the return value of "javafx.scene.layout.VBox.getScene()" is null
	at bisq.desktop.main.content.components.ChatMessagesListView$Controller.lambda$selectedChannelChanged$3(ChatMessagesListView.java:257)
	at bisq.desktop.common.threading.UIThread.run(UIThread.java:25)
	at bisq.desktop.main.content.components.ChatMessagesListView$Controller.selectedChannelChanged(ChatMessagesListView.java:228)
	at bisq.common.observable.Observable.addObserver(Observable.java:40)
	at bisq.desktop.main.content.components.ChatMessagesListView$Controller.onInTransitionCompleted(ChatMessagesListView.java:315)
	at bisq.desktop.main.content.components.ChatMessagesListView.onInTransitionCompleted(ChatMessagesListView.java:148)
	at bisq.desktop.main.content.components.ChatMessagesComponent$Controller.onInTransitionCompleted(ChatMessagesComponent.java:515)
	at bisq.desktop.main.content.components.ChatMessagesComponent.onInTransitionCompleted(ChatMessagesComponent.java:131)
	at bisq.desktop.main.content.chat.BaseChatController.onInTransitionCompleted(BaseChatController.java:113)
	at bisq.desktop.main.content.chat.BaseChatView.onInTransitionCompleted(BaseChatView.java:149)
	at bisq.desktop.common.view.TabView.lambda$setNewContent$7(TabView.java:139)
	at bisq.desktop.common.Transitions.lambda$fadeIn$0(Transitions.java:122)
	at javafx.animation.Animation.runHandler(Animation.java:1102)
	at javafx.animation.Animation.finished(Animation.java:1096)
	at javafx.animation.AnimationAccessorImpl.finished(AnimationAccessorImpl.java:49)
	at com.sun.scenario.animation.shared.SingleLoopClipEnvelope.timePulse(SingleLoopClipEnvelope.java:107)
	at javafx.animation.Animation.doTimePulse(Animation.java:1189)
	at javafx.animation.Animation$1.lambda$timePulse$0(Animation.java:207)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.animation.Animation$1.timePulse(Animation.java:206)
	at com.sun.scenario.animation.AbstractPrimaryTimer.timePulseImpl(AbstractPrimaryTimer.java:343)
	at com.sun.scenario.animation.AbstractPrimaryTimer$MainLoop.run(AbstractPrimaryTimer.java:266)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:571)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:555)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulseFromQueue(QuantumToolkit.java:548)
	at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$runToolkit$11(QuantumToolkit.java:353)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

* I also noticed some bizarre behaviour when loading the chat messages. This does not happen in main. See recording below.

![review-chat](https://github.com/bisq-network/bisq2/assets/145597137/45fb2160-314a-4c4b-b6d8-0699ada05bd1)

* When it comes to the refactor, and given that the code you are touching has been just introduced by me, I would appreciate if next time you give me your feedback in a review when I make the PR, rather than changing it afterwards. In the case that you didn't see it, mentioning me in the PR would also work, specially since I am currently actively working on this part of the code.

djing-chan commented:
@axpoems Thanks for the review.

> I am getting NPE 

Thanks for reporting. Fixed in commit e7fbcce

> I also noticed some bizarre behaviour when loading the chat messages. 

Seems the list get filled 2 times and the scrollpane gets set to the bottom position with a delay. I will look into that.

> When it comes to the refactor, and given that the code you are touching has been just introduced by me, I would appreciate if next time you give me your feedback in a review when I make the PR, rather than changing it afterwards. In the case that you didn't see it, mentioning me in the PR would also work, specially since I am currently actively working on this part of the code.

When doing more testing to find some observed bugs (e.g. private messages not arriving sometimes) I got into some issues with the current code base. I cannot say if those issues have been present already for longer or have been introduced by your recent changes. So I got my hand on those to fix that and in that process I saw that the hierarchy structure of the chat MVC classes was difficult to understand and tried to make it more clear. That led to a larger series of refactoring. The other thing I refactored was to rename the Channel class as it was a bit confusing to me if its a domain object or a view model, thus I renamed it to follow more the existing pattern.
To report the issues back to you and waiting for a fix would have blocked me in further investigating the bugs and it was also not clear initially what the issues are, so it took me some dev work to figure out.

djing-chan commented:
@namloan Could you have a look to https://github.com/bisq-network/bisq2/pull/1533/commits/b939be789bb35d2e0165d1facb6fd3712e28c5d9 ?

djing-chan commented:
@axpoems Could you test/review again? The scroll behaviour is now the same as in main.
But we still have another problem. See https://github.com/bisq-network/bisq2/issues/1538

HenrikJannsen reviewed (approved):
ACK
